static/elements/chromedash-upcoming.js

```diff
@@ -52,12 +52,110 @@ class ChromedashUpcoming extends LitElement {
       loginUrl: {type: String},
       starredFeatures: {type: Object}, // will contain a set of starred features
       cardWidth: {type: Number},
+      lastFetchedOn: {type: Number},
+      lastMilestoneVisible: {type: Number},
+      milestoneArray: {type: Array},
+      milestoneInfo: {type: Object},
+      cardsToFetchInAdvance: {type: Number},
     };
   }
 
+  set lastFetchedOn(val) {
+    let oldVal = this._lastFetchedOn;
+    this._lastFetchedOn = val;
+    this.fetchNextBatch().then(()=>{
+      this.requestUpdate('lastFetchedOn', oldVal);
+    }).catch(() => {
+      alert('Some error occurred. Please refresh the page or try again later.');
+    });
+  }
+
+  get lastFetchedOn() {
+    return this._lastFetchedOn;
+  }
+
   constructor() {
     super();
     this.cardWidth = this.computeWidthofCard();
+    this.starredFeatures = new Set();
+    this.milestoneArray = [];
+    this.milestoneInfo = {};
+    this.cardsToFetchInAdvance = 3;
+  }
+
+  async fetchNextBatch() {
+    // promise to fetch next milestones
+    const nextMilestones = window.csClient.getSpecifiedChannels(this._lastFetchedOn+1+1,
+      this._lastFetchedOn+1+this.cardsToFetchInAdvance);
+
+    let tempMilestoneArray = [];
+    for (let i = 1; i <= this.cardsToFetchInAdvance; i++) {
+      tempMilestoneArray.push(this._lastFetchedOn+1+i);
+    }
+
+    // promise to fetch features in next milestones
+    let milestoneFeaturePromise = {};
+    tempMilestoneArray.forEach((milestone) => {
+      milestoneFeaturePromise[milestone] = window.csClient.getFeaturesInMilestone(milestone);
+    });
+
+    let tempMilestoneInfo;
+    let milestoneFeatures = {};
+
+    try {
+      tempMilestoneInfo = await nextMilestones;
+
+      for (let milestone of tempMilestoneArray) {
+        milestoneFeatures[milestone] = await milestoneFeaturePromise[milestone];
+      }
+    } catch (err) {
+      throw (new Error('Unable to load Features'));
+    }
+
+
+    // add some details to milestone information fetched
+    tempMilestoneArray.forEach((milestone) => {
+      tempMilestoneInfo[milestone].version = milestone;
+      tempMilestoneInfo[milestone].features =
+          this.mapFeaturesToShippingType(milestoneFeatures[milestone]);
+    });
+
+    // update the properties to render the latest milestone cards
+    this.milestoneInfo = Object.assign({}, this.milestoneInfo, tempMilestoneInfo);
+
+    this.milestoneArray = [...this.milestoneArray, ...tempMilestoneArray];
+  }
+
+  mapFeaturesToShippingType(features) {
+    const featuresMappedToShippingType = {};
+    features.forEach(f => {
+      const component = f.browsers.chrome.status.text;
```

Comment:
"component" is probably left over from the old schedule page, try using variable name "shippingType".

Reply:
Oops... I must have left it by mistake. Thanks for pointing it out. 

Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/c8b6cdc9e8276fb78e5ab4babe3631140318c908) commit